### PR TITLE
[Filesystem] silence fileperms() issues on remote filesystems

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23312
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -69,7 +69,9 @@ public function copy($originFile, $targetFile, $overwriteNewerFiles = false)
             }
 
             // Like `cp`, preserve executable permission bits
-            @chmod($targetFile, fileperms($targetFile) | (fileperms($originFile) & 0111));
+            if (stream_is_local($originFile)) {
```
Comment (xabbuh) [inline src/Symfony/Component/Filesystem/Filesystem.php]: I just wonder if there are other resource types we need to consider too

Comment (javiereguiluz) [inline src/Symfony/Component/Filesystem/Filesystem.php]: I was reading about the `stream_is_local()` function and I saw this user comment: http://php.net/manual/en/function.stream-is-local.php#121126 Should we take that into account?

Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: TL;DR @javiereguiluz's point: stream_is_local return false for `file://` paths, but chmod works on them. So yes, we should have a case for them.

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ public function copy($originFile, $targetFile, $overwriteNewerFiles = false)
             }
 
             // Like `cp`, preserve executable permission bits
-            @chmod($targetFile, fileperms($targetFile) | (fileperms($originFile) & 0111));
+            @chmod($targetFile, fileperms($targetFile) | (@fileperms($originFile) & 0111));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: shouldn't the first "fileperms" also be silenced?
would that work?
`@(chmod($targetFile, fileperms($targetFile) | (fileperms($originFile) & 0111))));`

Comment (xabbuh) [inline src/Symfony/Component/Filesystem/Filesystem.php]: Shouldn't we always be able to get the perms of the newly written file?

Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: can't the new file be remote, thus have the same issue?

**Comment by nicolas-grekas** [comment]
wait, the original issue is buggy! things are already silenced, that's a "SilencedErrorContext"

**Pull request closed without merging**